src/fireedge/src/modules/components/Charts/MultiChart/helpers/subComponents/ChartRenderer.js

```diff
@@ -0,0 +1,306 @@
+
+import { Tr } from '@modules/components/HOC'
+import DataGridTable from '@modules/components/Tables/DataGrid'
+import { useTheme } from '@mui/material'
+import { CustomTooltip } from '@modules/components/Tooltip'
+import { sentenceCase } from '@UtilsModule'
+
+import {
+  generateColorByMetric,
+  GetChartConfig,
+  GetChartElementConfig,
+  CustomXAxisTick,
+} from '@modules/components/Charts/MultiChart/helpers/scripts'
+import {
+  FormatPolarDataset,
+  PolarTooltip,
+} from '@modules/components/Charts/MultiChart/helpers/subComponents'
+
+const CHART_TYPES = {
+  BAR: 'bar',
+  LINE: 'line',
+  AREA: 'area',
+  TABLE: 'table',
+  STACKED_BAR: 'stackedBar',
+  RADIAL_BAR: 'radialBar',
+}
+
+const ChartComponents = {
+  CARTESIAN: {
+    [CHART_TYPES.BAR]: BarChart,
+    [CHART_TYPES.STACKED_BAR]: BarChart,
+    [CHART_TYPES.LINE]: LineChart,
+    [CHART_TYPES.AREA]: AreaChart,
+    [CHART_TYPES.TABLE]: DataGridTable,
+  },
+  POLAR: {
+    [CHART_TYPES.RADIAL_BAR]: RadialBarChart,
+  },
+}
+
+const ChartElements = {
+  CARTESIAN: {
+    [CHART_TYPES.BAR]: Bar,
+    [CHART_TYPES.STACKED_BAR]: Bar,
+    [CHART_TYPES.LINE]: Line,
+    [CHART_TYPES.AREA]: Area,
+  },
+  POLAR: {
+    [CHART_TYPES.RADIAL_BAR]: RadialBar,
+  },
+}
+
+/**
+ * Renders a chart based on the provided type and data.
+ *
+ * @param {object} props - The properties for the component.
+ * @param {'bar' | 'line' | 'area' | 'table'} props.chartType - The type of chart to render.
+ * @param {Array} props.datasets - The datasets to be used for the chart.
+ * @param {object} props.selectedMetrics - The metrics selected for display.
+ * @param {Function} props.customChartDefs - Custom definitions for the chart.
+ * @param {Array} props.paginatedData - The paginated data for the chart.
+ * @param {Array} props.tableColumns - The columns for the table chart type.
+ * @param {Function} props.humanReadableMetric - Function to convert metric keys to human-readable format.
+ * @param {string} props.groupBy - The variable to group data under.
+ * @param {object} props.metricHues - Object containing hue values for different metrics.
+ * @param {boolean} props.disableLegend - Disables the legend underneath the charts.
+ * @param {string} props.coordinateType - Cartesian or Polar coordinate system.
+ * @param {Function} props.onElementClick - Callback to handle element click event.
+ * @returns {React.Component} The rendered chart component.
+ */
+export const ChartRenderer = ({
+  chartType,
+  datasets,
+  selectedMetrics,
+  customChartDefs,
+  paginatedData,
+  tableColumns,
+  humanReadableMetric,
+  groupBy,
+  coordinateType,
+  metricHues,
+  disableLegend,
+  onElementClick,
+}) => {
+  const ChartComponent = ChartComponents[coordinateType][chartType]
+  const ChartElement = ChartElements[coordinateType][chartType]
+  const theme = useTheme()
+
+  // Map with translation for each metric
+  const translationMap = mapValues(selectedMetrics, (_, key) => {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).